dspace-api/src/main/java/org/dspace/core/DBConnection.java

```diff
@@ -13,48 +13,121 @@
 import java.sql.SQLException;
 
 /**
- * Interface representing a Database connection, this class should only be access by the context object.
+ * Interface representing a persistence provider "session".
+ * Implementations will wrap something like a JPA EntityManager or a Hibernate
+ * Session.  The terms "database" and "connection" are historic and have no
+ * direct relationship to a JDBC Connection or even a connection pool.
+ *
+ * <p> This class should only be accessed by an enclosing {@link Context} object.
  *
  * @author kevinvandevelde at atmire.com
  * @param <T> class type
  */
 public interface DBConnection<T> {
 
+    /**
+     * Access to the underlying session object.
```

Comment:
While I know you already clarified the meaning of "session" at the top of this class, I'm not sure if we should (re)clarify in this method that "session" is a persistence provider session (and has no relation to HTTPSession). 

If you think this is obvious (even to newbies), feel free to ignore this note.